nca/CoreDS/Peer.py

```diff
@@ -460,6 +456,48 @@ def is_ipv4_block(self):
         return cnt == len(self.interval_set)
 
 
+class DNSEntry(Peer):
+    """
+    represents DNS entries, produced by ServiceEntry objects
+    """
+    def __init__(self, host_mindfa=None, name=None, namespace=None):
+        """
+        Constructs a DNSEntry from the host minDFA provided
+        :param MinDFA host_mindfa: represents the minDFA of the host regex
+        """
+        Peer.__init__(self, name, namespace)
+        self.host_mindfa = host_mindfa
```

Comment:
why do we need `MinDFA` object and not just use simple regex matching?

Reply:
an example case behind using the host_mindfa is:

> suppose a `ServiceEntry` has a host named `*.wikipedia.org`
> and a Sidecar enables egress to a host named `en.wikipedia.org` 
> when parsing the Sidecar, I need to check if we have a DNSEntry peer that is equal or "contains"
 the host dns name `en.wikipedia.org`  in order to add it to the sidecar's workload peers connections 

so, I used the  `peer.host_mindfa.__contains__(host_dns)` to check that . (as it exists in our code)
Do you think, it is better to replace this check with a simple regex matching one?

Reply:
yes.
There is an overhead to build a mindfa object, and we use it for the canonical representation of regex connection attributes.. but in this case if we can use simple regex matching it is preferred.

Reply:
okay, I'll replace then, thanks

Reply:
done